### PR TITLE
Fix full system error due to `isConvex` failure

### PR DESCRIPTION
**Pull request by jsn0716**
### Description

Fixes error caused due to operation with `Field` while it does not have valid dimensions at the start up of full system.

More detailed steps on how the error occurs:
1. The `NetworkFilter` creates a `Field` with all 0 arguments expecting `getFilterData` to be called for access to `Field`, which would update with given protobuf data
2. `NetworkClient` receives packet, but it has no geometry data
3. The same packet has detection, so it tries to call `getFilteredBallData`
4. The new detection is checked to see if it is inside the `Field`, which does so by creating a `Rectangle`, which is a `ConvexPolygon`
5. Error because the `Field` still has all 0 dimensions

### Testing Done

Running full system does not crash anymore

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by jsn0716** [comment]
I put the `Field` validity check (minimal check that fixes the problem at the moment) on `NetworkClient` because it tries to update other `World` parameters that may use an invalid `Field`; however, it might be appropriate for the `NetworkClient` to do minimal error checking, and left for `Backend` to handle any invalid `Field` (or maybe `World` in general). Let me know what you think @MathewMacDougall @garethellis0 @scveloso

**Review by MathewMacDougall — DISMISSED** [review]
LGTM. I've created #1218 to move this fix to SensorFusion once it's integrated, since that's ultimately where this should belong (as a "filter"). Long-term I also don't think we should expose `isValid()` functions in our `World` datastructures because ideally by the time we publish a `World`, everything that consumes it should be able to assume it's valid. This will be more easily addressed in SensorFusion as well once the filters are integrated.

File: src/software/backend/input/network/networking/network_client.cpp [hunk 1]
```diff
@@ -129,59 +129,65 @@ void NetworkClient::filterAndPublishVisionData(SSL_WrapperPacket packet)
...
     {
-        SSL_DetectionFrame detection = *packet.mutable_detection();
-        bool camera_disabled         = false;
-
-        // We invert the field side if we explicitly choose to override the values
-        // provided by refbox. The 'defending_positive_side' parameter dictates the side
-        // we are defending if we are overriding the value
-        if (refbox_config->OverrideRefboxDefendingSide()->value() &&
-            refbox_config->DefendingPositiveSide()->value())
+        if (packet.has_detection())
```
Comment (garethellis0) [inline src/software/backend/input/network/networking/network_client.cpp:134]: Do we need to nest these `if`'s or can we write it as `if (world.field().isFieldValid() && packet.has_dectection())`?

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/software/backend/input/network/networking/network_client.cpp (hunk 1 above)
Comment (jsn0716) [reply src/software/backend/input/network/networking/network_client.cpp:134]: We would need to nest them for `received_world_callback(world);` to be called when there's no detection but `Field` is valid

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Comment by jsn0716** [comment]
@MathewMacDougall I added the missing javadoc comment to `isFieldValid`, going to need reapproval

**Review by MathewMacDougall — APPROVED** [review]
:+1:

**Pull request merged**